langserve/server.py

```diff
@@ -464,31 +477,44 @@ async def stream_log(
 
         async def _stream_log() -> AsyncIterator[dict]:
             """Stream the output of the runnable."""
-            async for chunk in runnable.astream_log(
-                input_,
-                config=config,
-                diff=True,
-                include_names=stream_log_request.include_names,
-                include_types=stream_log_request.include_types,
-                include_tags=stream_log_request.include_tags,
-                exclude_names=stream_log_request.exclude_names,
-                exclude_types=stream_log_request.exclude_types,
-                exclude_tags=stream_log_request.exclude_tags,
-            ):
-                if not isinstance(chunk, RunLogPatch):
-                    raise AssertionError(
-                        f"Expected a RunLog instance got {type(chunk)}"
-                    )
-                data = {
-                    "ops": chunk.ops,
-                }
-
-                # Temporary adapter
+            try:
+                async for chunk in runnable.astream_log(
+                    input_,
+                    config=config,
+                    diff=True,
+                    include_names=stream_log_request.include_names,
+                    include_types=stream_log_request.include_types,
+                    include_tags=stream_log_request.include_tags,
+                    exclude_names=stream_log_request.exclude_names,
+                    exclude_types=stream_log_request.exclude_types,
+                    exclude_tags=stream_log_request.exclude_tags,
+                ):
+                    if not isinstance(chunk, RunLogPatch):
+                        raise AssertionError(
+                            f"Expected a RunLog instance got {type(chunk)}"
+                        )
+                    data = {
+                        "ops": chunk.ops,
+                    }
+
+                    # Temporary adapter
+                    yield {
+                        "data": simple_dumps(data),
+                        "event": "data",
+                    }
+                yield {"event": "end"}
+            except BaseException:
                 yield {
-                    "data": simple_dumps(data),
-                    "event": "data",
+                    "event": "error",
+                    # Do not expose the error message to the client since
+                    # the message may contain sensitive information.
+                    # We'll add client side errors for validation as well.
+                    "data": json.dumps(
+                        {"status_code": 500, "message": "Internal Server Error"}
+                    ),
                 }
-            yield {"event": "end"}
+                yield {"event": "end"}
```

Comment:
Should `error` imply an `end`?

Reply:
Wasn't sure about that -- we can have it imply end or always send an end event. Any preferences? 

b/c older clients will break anyway, neither one is backwards compatible